server/item/written_book.go

```diff
@@ -0,0 +1,75 @@
+package item
+
+// WrittenBook is the item created after a book and quill is signed. It appears the same as a regular book, but
+// without the quill, and has an enchanted-looking glint.
+type WrittenBook struct {
+	// Title is the title of the book.
+	Title string
+	// Author is the author of the book.
+	Author string
+	// Generation is the copy tier of the book. 0 = original, 1 = copy of original,
+	// 2 = copy of copy.
+	Generation WrittenBookGeneration
+	// Pages represents the pages within the book.
+	Pages []string
+}
+
+// MaxCount always returns 1.
+func (w WrittenBook) MaxCount() int {
+	return 1
+}
+
+// Page returns a specific page from the book and true when the page exists. It will otherwise return an empty string
+// and false.
+func (w WrittenBook) Page(page int) (string, bool) {
+	if page < 0 || len(w.Pages) <= page {
+		return "", false
+	}
+	return w.Pages[page], true
+}
+
+// DecodeNBT ...
+func (w WrittenBook) DecodeNBT(data map[string]any) any {
+	if pages, ok := data["pages"].([]any); ok {
+		w.Pages = make([]string, len(pages))
+		for i, page := range pages {
+			w.Pages[i] = page.(map[string]any)["text"].(string)
+		}
+	}
+	if v, ok := data["title"].(string); ok {
+		w.Title = v
+	}
```

Comment:
Can change this to `w.Title, _ = data["title"].(string)`. Same for the author below.